minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/util/ConfigTransformer.java

```diff
@@ -546,6 +550,37 @@ protected static void addConnection(final Element parentElement, ConnectionSchem
         }
     }
 
+    protected static String getConnectionName(ConnectionSchema connectionProperties) {
+        String connectionName = connectionProperties.getName();
+        if (StringUtil.isNullOrEmpty(connectionName)) {
+            return EMPTY_NAME;
+        }
+        return connectionName;
+    }
+
+    /**
+     * Will replace all characters not in [A-Za-z0-9_] with _
+     *
+     * This has potential for collisions so it will also append numbers as necessary to prevent that
+     *
+     * @param ids id map of already incremented numbers
+     * @param name the name
+     * @return a unique filesystem-friendly id
+     */
+    protected static String getUniqueId(Map<String, Integer> ids, String name) {
+        String baseId = name.replaceAll("[^A-Za-z0-9_]", "_");
+        String id = baseId;
+        Integer idNum = ids.get(baseId);
+        while (ids.containsKey(id)) {
+            id = baseId + "_" + idNum++;
+        }
+        if (id != baseId) {
+            ids.put(baseId, idNum);
+        }
+        ids.put(id, 2);
```

Comment:
if we opt to keep the current approach, not sure I follow this.  Was this a remnant of testing?